autogen/agentchat/conversable_agent.py

```diff
@@ -822,66 +822,51 @@ def _summarize_chat(
         """
         agent = self if agent is None else agent
         summary = ""
-        if method == "last_msg":
-            try:
-                summary = agent.last_message(self)["content"]
-                summary = summary.replace("TERMINATE", "")
-            except (IndexError, AttributeError):
-                warnings.warn("Cannot extract summary from last message.", UserWarning)
-        elif method == "reflection_with_llm":
+        if method == "reflection_with_llm":
             prompt = ConversableAgent.DEFAULT_summary_prompt if prompt is None else prompt
             if not isinstance(prompt, str):
                 raise ValueError("The summary_prompt must be a string.")
             msg_list = agent._groupchat.messages if hasattr(agent, "_groupchat") else agent.chat_messages[self]
```

Comment:
I should have noted this with the previous PR, but it's a bit awkward to be anticipating a subclass here and reaching into a private subclass attribute.

It may also break composability -- but I'm not sure. @ekzhu  or @sonichi to comment.

Finally, these messages will be used in self._generate_oai_reply_from_client in this solution. However, the groupchat.messages is a bit of a different message structure than is typical (for one thing, "role" might be a little confusing). Are we sure _generate_oai_reply_from_client is completely compatible with such message lists?

Reply:
> it's a bit awkward to be anticipating a subclass here and reaching into a private subclass attribute.

In this case we could have a property of ConversableAgent class (GroupChatManager is a ConversableAgent) called `messages_for_reflection` that is overridden by GroupChatManager to hide away the logic of exporting message list.

Though generally having to do this could mean that the GroupChatManager is assigned with too many responsibilities. In the future design we can consider breaking it apart into regular ConversableAgent.